### PR TITLE
Replace @jupyterlab/rjsf with FormComponent from @jupyterlab/ui-components

### PR DESCRIPTION
**Pull request by Meriem-BenIsmail**
### Description
This PR updates the **JupyterCAD** to use the [FormComponent](packages/ui-components/src/components/form.tsx)   from the `@jupyterlab/ui-components` package instead of [SchemaForm](https://github.com/deathbeds/jupyterlab-starters/tree/main/packages/jupyterlab-rjsf).
### Key Changes:
1. **Form Component Migration**:
    - Replaced [SchemaForm](https://github.com/deathbeds/jupyterlab-starters/tree/main/packages/jupyterlab-rjsf) of `@jupyterlab/rjsf` with [FormComponent](packages/ui-components/src/components/form.tsx), the standardized form component in the JupyterLab ecosystem.
2. **Validator Integration**:
    - Integrated [validator-ajv8](https://www.npmjs.com/package/@rjsf/validator-ajv8) with `FormComponent` to ensure schema validation.
3. **Custom Array Handling**:
    - Added a custom array field (`CustomArrayFieldWithoutButtons`) to maintain existing customization for array data entry.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,5 +77,8 @@
     "stylelint-prettier": "^4.0.0",
     "typescript": "^5",
     "webpack": "^5.76.3"
+  },
+  "dependencies": {
+    "@jupyterlab/ui-components": "^4.3.1"
```
Comment (martinRenou) [inline package.json:82]: We should move this dependency under the `packages/base/package.json` instead

File: packages/base/src/panelview/formbuilder.tsx [hunk 2]
```diff
@@ -23,16 +23,72 @@ interface IProps {
...
+      }}
+      onSubmit={onSubmit}
+    />
+  );
+};
 
 // Reusing the datalayer/jupyter-react component:
 // https://github.com/datalayer/jupyter-react/blob/main/packages/react/src/jupyter/lumino/Lumino.tsx
-export const LuminoSchemaForm = (
+export const LuminoForm = (
```
Comment (martinRenou) [inline packages/base/src/panelview/formbuilder.tsx:85]: Should we get rid of this `LuminoForm`? I understand we don't need it anymore?

**Review by martinRenou — COMMENTED** [review]
Thanks for starting this! I have some early comments/suggestions

**Comment by martinRenou** [comment]
The Union operator form does not show dropdown selectors anymore:

# Before this PR:

https://github.com/user-attachments/assets/5c65f4ab-faa3-4df1-b34c-808f5be5ce2e

# With this PR:

https://github.com/user-attachments/assets/a8d6df29-3dff-4675-9e8b-8feeed5f5d07

**Comment by Meriem-BenIsmail** [comment]
please update snapshots

**Comment by Meriem-BenIsmail** [comment]
please update snapshots

**Comment by arjxn-py** [comment]
Triggered CI

File: packages/base/src/panelview/formbuilder.tsx [hunk 3]
```diff
@@ -154,59 +143,70 @@ export class ObjectPropertiesForm extends React.Component<IProps, IStates> {
...
+      return <div>{this.buildForm()}</div>;
+    }
+    return (
+      <div
+        className="jpcad-property-panel"
+        data-path={this.props.filePath ?? ''}
+      >
+        <div className="jpcad-property-outer jp-scrollbar-tiny">
+          <WrappedFormComponent
+            schema={schema}
```
Comment (martinRenou) [inline packages/base/src/panelview/formbuilder.tsx:160]: Maybe?

```suggestion
            schema={{...schema}}
```

**Comment by Meriem-BenIsmail** [comment]
please update snapshots

**Comment by Meriem-BenIsmail** [comment]
please update snapshots

**Review by martinRenou — APPROVED** [review]
Thanks 🚀 !

**Comment by martinRenou** [comment]
please update snapshots

**Pull request merged**